packages/firestore/src/lite-api/reference.ts

```diff
@@ -78,12 +78,9 @@ export type WithFieldValue<T> = T extends Primitive
  */
 export type UpdateData<T> = T extends Primitive
   ? T
-  : T extends Map<infer K, infer V>
```

Comment:
Why do you not need to handle `Map`s separately?

Reply:
`UpdateData` shouldn't allow `Maps` to begin with because it was designed to take `{ [fieldPath: string]: any}` as an input originally. `WithFieldValue` and `PartialWithFieldValue` both don't allow `Map`s.